core/llm.go

```diff
@@ -319,64 +324,75 @@ func (r *LLMRouter) Route(model string) (*LLMEndpoint, error) {
 
 func (r *LLMRouter) LoadConfig(ctx context.Context, getenv func(context.Context, string) (string, error)) error {
 	if getenv == nil {
-		getenv = func(ctx context.Context, key string) (string, error) {
```

Comment:
what's the full intent of these changes? parallelize, obviously, refactor a little, but does this change error output at all?

sidenote: why do you pass ctx to roundly ignore it?

Reply:
The goal is just to parallelize, since I saw CI was taking a very long time to chew through all of these one by one. There should be no behavior difference (besides speed).

`ctx` is only ignored by the getenv fallback function (`if getenv == nil`) - afaik it's not ignored by the "real" getter that's passed in.

Reply:
ah, i read right past the `getenv func` param, now this makes sense